### DISCUSSION TITLE
Being able to use the mobile app offline or standalone

### DISCUSSION THREAD

**Discussion by Lendi24** [immich-app/immich#4466] [Feature Request]
One thing I really like about google photos is how it can replace the default photos app entirely 

I feel like there are only few things missing for this type of experience here. If you could only see you local pictures next the the remote ones and use the app without logging in, we would basically be there! 

This could allow for some interesting use-cases, like options to automatically offload local copies for images newer than one year or only keeping low resolution versions locally.

One could also see cerian albums being marked as "both on server and device", keeping them in sync with other devices while still being easily accessible! 

This may not be something to focus on this early, but it would be a really cool feature! 
Also, it's my first time writing feature requests, so I'm sorry if I messed something up! 😅

**Comment by raelgc** [comment]
Basically the "offline" copies could be initially, like on Google Photos, just the content from `/DCIM/Camera`. When you select a non `/DCIM/Camera` picture, you have the option to download it, which will download it exactly to `/DCIM/Camera`.

**Reply by dev-ritik** [reply]
This sounds pretty cool feature to have. The app currently is unusable if it is offline. Do we have plans to get it implemented? Happy to help with the dev work as well

**Reply by alextran1502** [reply]
It is usable to some extended. If the remote images are not cached, it won't show if you don't have connection to the server though

**Reply by crashmit** [reply]
Maybe I've been missing it but what I'd love to see is an option enable automatic caching of all images on the device. Perhaps with quality as a selection for the user so it doesn't need to be the full res one if you're on limited phone storage. This could also help the UX when on slower internet connections as something is always already visible and in all circumstances the library is browsable. As @Lendi24 said, that would also help me to use Immich as my sole photos app on my phone.

**Comment by not4smurf** [comment]
I'm not so concerned about old images being cached.  The problem I have with the Android app when offline is that I can't use it to look at photos I've just taken.  If I open the app when not connected to the server it just spins.  I think it will timeout eventually and allow access to my local photos, but I don't wait for it - I just kill it and switch back to Google photos, waste a little while figuring out the (intentionally?) horrible UI Google presents when it's set to not backup, and then look through my images.

It seems that the app launches, tries to connect and then falls back to local only mode.  I'd like to see it default to local-only mode and attempt server connection in the background - I should only be aware that I'm offline if I try to access something that isn't stored locally.

**Reply by alextran1502** [reply]
@not4smurf yeah we are working toward a better support for offline mode so you can use Immich as the viewer regardless of your server connection

**Reply by palitu** [reply]
@alextran1502 

> @not4smurf yeah we are working toward a better support for offline mode so you can use Immich as the viewer regardless of your server connection

Ah perfect.  i actually just came to ask about the App.  It is unusable as a photo viewing app, so my wife keeps using google photos, and i use fossify gallery. 

I would definitely loove to see the app be a photo viewer, as it also adds in the photos that are only on the server.  As a minimum local caching for thumbnails would be a minimum, using the local photos if available, and have a limited cache of online only photos, that are stored and purged dynamically.

But all of this is not going to get used if it take 10s for the app to open.

For a little bit more context, i live in a country that has limited internet, and it is expensive, so i spend a lot of time offline.

Thanks! i have been using and loving immich since pretty close to the start, and it just keeps going up and up!

**Reply by fedeoeae** [reply]
> @not4smurf yeah we are working toward a better support for offline mode so you can use Immich as the viewer regardless of your server connection

Just for information, how far is this away? Weeks/months/years?

**Reply by alextran1502** [reply]
We don't do ETA. Probably months

**Reply by Hear7beaT** [reply]
> We don't do ETA. Probably months

Is there an update on this general idea of being able to access local files if there is no connection to the server without waiting for 10 seconds? 

Thanks

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by thomas-hauchecorne** [reply]
> https://justinmayer.com/posts/any-updates/

> Do people really think progress happens somewhere else, behind the scenes? The probability of some out-of-sight meeting, sprint, or planning session happening without being visible somewhere in the project’s issue tracker is almost zero.

In the case of Immich, yes there are meetings and planning happening out-of-sight.
I'm not arguing in favor of posting "any updates?", but your link doesn't really apply here.

**Reply by bo0tzz** [reply]
> yes there are meetings and planning happening out-of-sight.

Not nearly as much as you probably think; yes we do coordinate internal to the team, but our planning, if any, is rarely more than a week or so ahead of anything happening on github.

**Comment by quimodotcom** [comment]
bump. mainly cause i could fully degoogle my photos app entirely.

**Reply by bo0tzz** [reply]
Please don't bump threads. If you want to support this feature request you can upvote the original post.

**Comment by 1661410107114145152525** [comment]
Thanks for pointing my feature request (https://github.com/immich-app/immich/discussions/21017) to this @bo0tzz :). As the request originated 2023, since got many upvotes and still is open I wonder how high this is on your priorities list, if it is still years or something we could expect to see tackled in a sooner rather than later future? (no need to justify I can totally relate that there have been many really important features and fixes, I just try to understand if I can realistically expect it in a neater future of if this is so complex or irrelevant that it will propably still be open for years)

**Reply by bo0tzz** [reply]
Full offline support is complex and pretty hard, but we're always moving towards it.

**Reply by 1661410107114145152525** [reply]
Addition to my intitial post above: @bo0tzz Oh and as I saw the link above - this was not meant as a typical "any updates" request, just needed some information so I know if I need to workaround it, drop Immich or if waiting pays out if those use cases occur more often. And yes I already donated even before having any feature requests in mind as you guys are doing awesome work and you deserve every cent, but I would never expect any feature to progress faster due to a donation (except I am a large company sponsor which I am not, so all that's left is asking).

**Reply by 1661410107114145152525** [reply]
> Full offline support is complex and pretty hard, but we're always moving towards it.

Sounds great! @bo0tzz So it is kept in mind and something that's more about technical duration, not priorities per se. Thanks! :)

Btw: Please keep on mind that we would also benefit from intermediate results in between. E.g. album specific caching as follow-up or so.

**Reply by gubbel-alt** [reply]
I'm really looking forward to the offline functionality of the app (iOS) and hope it won't be too much longer. Thank you for your tireless work!

**Comment by sibaar** [comment]
Second this! Please make this happen.

**Comment by lionking102** [comment]
Offline support for the app would be amazing if you only access immich through VPN and don`t always want VPN to be enabled.

**Comment by defiling9046** [comment]
Would this be important enough to add to the end of the roadmap on the Immich website @bo0tzz ? Either way thank you for all the amazing work bo0tzz and the rest of the Immich team!🎉

**Comment by antoniovj1** [comment]
I would love to see the app load quickly even without an internet connection, so it can be reliably used as a local gallery.

For example, when I'm hiking in the mountains, I often don’t have internet access. In those situations, I still want to review the photos I’ve just taken, but the app doesn’t load properly without connectivity. As a result, I have to switch to Google Photos instead.

It would be great if the app could function smoothly offline, allowing users to browse and view locally stored photos without needing an active connection. This would make it possible to rely on the app as a primary gallery, even in offline scenarios.

Thanks a lot!

**Comment by RobinCoding** [comment]
First of all, thank you for this amazing project!
I’ve read through several posts and discussions about this feature, and local photo saving is something that’s not only for me personally very important but also for a lot others. That’s why I’d be happy to invest my time in contributing to the Flutter app to help make it happen.
I’m an experienced software developer/engineer myself, so I’m confident I could get up to speed with the codebase. I just wanted to reach out first to check whether this is something the team would welcome a contribution on, and whether there’s a preferred approach or any existing groundwork I should be aware of before I dive in. @bo0tzz  @alextran1502 
Thanks again for all the work you’ve put into this project!

**Reply by george1612** [reply]
Thank you very much, this sounds really promising! :) Have you also seen the issue linked by bo0tzz https://github.com/immich-app/immich/discussions/21017 and what are your thoughts here?

**Reply by RobinCoding** [reply]
Update: I was working on this and even had a first working draft but the creators of the project don't see this festure in their vision of the app.

**Reply by fedeoeae** [reply]
Walt, really? Which part of it? It seems like a basic feature to me. The app, as it is now, is basically only an interface for backing up. It's basically useless to actually view photos, even while online. It crashes often when looking through months where I took many pictures (like people do, when on vacation).

I guess it could be fixed by improving performance as the problem doesn't seem to exist when accessing the pictures on pc directly through the selfhosted website, but caching _some_ pictures in lower quality seems like it would be a huge step forward in usability.

The _very basic_ use case of wanting to show pictures of a recent vacation to others is enough to bring the app to it's knees. If they really don't want the app to be used to actually view the pictures, why did they even bother making them visible. They should have just made an interface for the backup process.
Well, maybe that's just my frustration speaking.

Thank you for the update anyway.

**Reply by gubbel-alt** [reply]
Thanks a lot for implementing a preview version. Is there anywhere I can test it? I’m really keen on this feature, too, since I have several terabytes of photos and videos. The reason I rarely look at them is that they’re stored on my NAS, which lacks a good mobile photo app; I prefer viewing images spontaneously without having to fire up my PC—I’m just too lazy for that. Since I started using Immich, I’ve been looking at my old photos much more often, and I’m really grateful for that. Stability with large libraries has also steadily improved; initially, I had to wait a long time for images to load, but it’s much faster now. Still, it would be a dream to have all my photos available offline in low resolution, similar to how Apple Photos handles it. After all, the thumbnails are already rendered on the server. I really hope this feature officially arrives at some point. Judging by the release notes for version 3, things seem to be moving in the right direction, at least for Android. We’ll see how it plays out for iOS.

**Comment by w00kee** [comment]
Regarding this we are basically almost there. If I have no internet connection at all e.g. WLAN and Mobile data are turned off, the app already loads in offline mode basically instantly (really short wait). **Please make it such that this is the default behavior**, load the photos view with local images first before trying to establish a connection to the server.
As I need an active VPN connection to connect to my server (which I sadly can not turn on by default), I run into the following issue:
The app tries to establish a connection to the server before showing any pictures, as the connection takes ages to fail (VPN is not turned on) I can't view my local pictures until at some point the app realizes that the server is not reachable.
Just make the app show local pictures before it tries to connect to the server.

**Reply by defiling9046** [reply]
@w00kee just so I can make sure, if i do not wish to make myself an immich server i can still enjoy using the mobile immich app as a completely local offline gallery app?

**Reply by bo0tzz** [reply]
No

**Reply by defiling9046** [reply]
> No

Ok I guess I just read "standalone" as something different in my head. Thanks for the clarification though.

**Comment by DD4711** [comment]
I would also strongly support this feature.

I self-host Immich on a Synology NAS. The current online experience is great with an excellent internet-connection, but offline browsing is one of the few remaining areas where Apple Photos and Google Photos still have an advantage.

This would make Immich much more useful while travelling, on airplanes, or in areas with poor mobile coverage, while requiring only a fraction of the storage needed for full offline synchronization.

Thank you for considering this feature and for all the amazing work on Immich!

**Comment by justspacedog** [comment]
I’d like to add another use case:

I use the official Immich app on an iPad as a dedicated digital photo frame for a family member who doesn’t have internet access and only a very limited mobile data plan.

By the end of the month, the data allowance is often exhausted. At that point, they can’t see a single photo anymore because previously viewed images don’t seem to remain available locally.

A persistent image cache (or the option to cache an album locally) would make a huge difference for this use case. Otherwise, I have to upload all my photos twice (once to Immich and once to iCloud Photos) just so they don’t disappear from the photo frame.
